meson does not detect missing libgmock

causing build failure during `ninja` instead of earlier

```
ccache c++ -Itest/9f86d08@@create-config@exe -Itest -I../test -I../test/../src -Itest/.. -I../test/.. -I../test/../sr_wrapper -I../subprojects/spdlog-1.4.2/./include -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -std=c++17 -g -pthread -DSPDLOG_COMPILED_LIB  -MD -MQ 'test/9f86d08@@create-config@exe/create-config_create-config.cc.o' -MF 'test/9f86d08@@create-config@exe/create-config_create-config.cc.o.d' -o 'test/9f86d08@@create-config@exe/create-config_create-config.cc.o' -c ../test/create-config/create-config.cc
../test/create-config/create-config.cc:18:10: fatal error: gmock/gmock-matchers.h: No such file or directory
 #include <gmock/gmock-matchers.h>
          ^~~~~~~~~~~~~~~~~~~~~~~~
```

The [Debian package list](https://packages.debian.org/buster/amd64/libgmock-dev/filelist) mentions it exists.

However, here's what your configure shows vs mine:

```
Dependency [1mGTest[0m found: [1;32mYES[0m (prebuilt)
Dependency [1mGMock[0m found: [1;32mYES[0m (building self)
```

```
Run-time dependency GTest found: YES 1.8.1
Run-time dependency GMock found: YES 1.8.1
```

It seems that the libgmock-dev package has no .pc file *and* ships a .a, whereas the arch package ships a .pc file *and* a .so. I'll see if I can fix this.

--- a/test/meson.build
+++ b/test/meson.build
@@ -1,4 +1,8 @@
 gtest_dep = dependency('gtest')
+gmock_h = meson.get_compiler('cpp').has_header('gmock/gmock.h')
+if not gmock_h
+  error('gmock/gmock.h not found')
+endif
 gmock_dep = dependency('gmock')
 
 # we need to rethink this..

